samples/excel/85-preview-apis/data-types-formatted-number.yaml

```diff
@@ -0,0 +1,149 @@
+order: 1
+id: excel-data-types-formatted-number
+name: 'Data types: Formatted numbers'
+description: This sample shows how to set and get data types using the formatted number properties.
+host: EXCEL
+api_set:
+    ExcelApi: BETA (PREVIEW ONLY)
+script:
+    content: |
+        $("#set-formatted-number-date").click(() => tryCatch(setFormattedNumberDate)); 
+        $("#set-formatted-number-currency").click(() => tryCatch(setFormattedNumberCurrency)); 
+        $("#get-formatted-number").click(() => tryCatch(getFormattedNumber));
+
+        async function setFormattedNumberDate() {
+          await Excel.run(async (context) => {
+            // This method creates a formatted number data type,
+            // and sets the format of this data type as a date.
+
+            // Create a worksheet called "Sample" and activate it.
+            context.workbook.worksheets.getItemOrNullObject("Sample").delete();
+            const sheet = context.workbook.worksheets.add("Sample");
+            sheet.activate();
+            
+            // Get cell A1 and write a number formatted as a date to cell A1.
+            const dateRange = sheet.getRange("A1");
+            dateRange.valuesAsJson = [
+              [
+                {
+                  type: Excel.CellValueType.formattedNumber,
+                  basicValue: 32889.0,
+                  numberFormat: "m/d/yyyy"
+                }
+              ]
+            ];
+            await context.sync();
+          });
+        }
+
+        async function setFormattedNumberCurrency() {
+          await Excel.run(async (context) => {
+            // This method creates a formatted number data type,
+            // and sets the format of this data type as a currency.
+
+            // Get the active worksheet and a range on that sheet.
+            const sheet = context.workbook.worksheets.getActiveWorksheet();
+            const currencyRange = sheet.getRange("A2");
+
+            // Write a number formatted as currency to cell A2.
+            currencyRange.valuesAsJson = [
```

Comment:
What does it look like to set more than one cell at once? Is it really clunky or is there an optimization we could showcase?

Reply:
I can look into this and, if an optimization strategy for multiple cells exists, include it in my next Script Lab sample for data types (web image is up next). Does that sound reasonable?

Reply:
Yeah, that's a good plan. Really, this is more of a question for the Data Types feature in general than for this particular sample.